packages/amplify_authenticator/lib/amplify_authenticator.dart

```diff
@@ -216,24 +228,84 @@ export 'src/widgets/form_field.dart'
 /// )
 /// ```
 /// {@endtemplate}
+///
+/// ## Custom UI
+///
+/// {@template amplify_authenticator.custom_builder}
+/// The authenticator provides prebuilt widgets for each step
+/// of the authentication flow based on the amplify config for
+/// your app. Some customizations can be acheived by providing
+/// custom forms (see [signInForm] and [signUpForm]) or through
+/// theming. To fully customize the authenticator UI,
+/// you can provide a custom builder method.
+///
+/// This can be used to change the layout of the authenticator,
+/// add your app's logo to certain views, add completely custom
+/// form fields (such as a terms and conditions field), and much
+/// more.
+///
+/// The example below shows a custom sign up view with an app
+/// bar and a flutter logo.
+///
+/// ```dart
+/// return Authenticator(
+///   authenticatorBuilder: (context, state) {
+///     switch (state.currentStep) {
+///       case AuthenticatorStep.signIn:
+///         return Scaffold(
+///           appBar: AppBar(title: const Text('My App')),
+///           body: Padding(
+///             padding: const EdgeInsets.all(16),
+///             child: Column(
+///               children: [
+///                 // flutter logo
+///                 const Center(child: FlutterLogo(size: 100)),
+///                 // prebuilt sign in form from amplify_authenticator package
+///                 SignInForm(),
+///               ],
+///             ),
+///           ),
+///         );
+///       default:
+///         return null;
+///     }
+///   },
+///   child: MaterialApp(
+///     builder: Authenticator.builder(),
+///     home: const RouteA(),
+///   ),
+/// );
+/// ```
+/// {@endtemplate}
 class Authenticator extends StatefulWidget {
   /// {@macro amplify_authenticator.authenticator}
   Authenticator({
     Key? key,
-    SignInForm? signInForm,
-    SignUpForm? signUpForm,
-    ConfirmSignInNewPasswordForm? confirmSignInNewPasswordForm,
+    this.signInForm,
+    this.signUpForm,
+    this.confirmSignInNewPasswordForm,
     this.stringResolver = const AuthStringResolver(),
     required this.child,
     this.useAmplifyTheme = false,
     this.onException,
     this.exceptionBannerLocation = ExceptionBannerLocation.auto,
     this.preferPrivateSession = false,
+    this.initialStep = AuthenticatorStep.signIn,
+    this.authenticatorBuilder,
+    this.padding = const EdgeInsets.all(32),
```

Comment:
This is the padding that is used for each screen. I exposed padding as a configurable property. For custom UIs that only customize some of the screens (Sign In, but not Reset Password) customers may use a different padding on their custom screens. There will then be inconsistencies between the prebuilt screens and the custom ones. This was the case in some of the examples I built.

I think this should default to `EdgeInsets.all(16)` or `EdgeInsets.symmetric(horizontal: 16, vertical: 32)` because I think 16 is a more common horizontal padding. However, 32 is what we currently use. Material doesn't have super clear guidance on this, but their docs on [responsive layouts under "breakpoints"](https://material.io/design/layout/responsive-layout-grid.html#breakpoints), it does indicate that extra small screens (phones) should have a margin of 16px.

I am leaving this change out for now, but @haverchuck @dnys1 if you both agree I can include it.